app/src/main/java/org/wikipedia/history/HistoryFragment.java

```diff
@@ -110,9 +111,20 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         historyList.setAdapter(adapter);
 
         LoaderManager.getInstance(requireActivity()).initLoader(HISTORY_FRAGMENT_LOADER_ID, null, loaderCallback);
+        setUpScrollListener();
         return view;
     }
 
+    private void setUpScrollListener() {
+        historyList.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+                ((MainActivity) requireActivity()).updateToolbarElevation(historyList.canScrollVertically(SCROLL_DIRECTION_UP));
```

Comment:
Can't we just use `historyList.computeVerticalScrollOffset() != 0`, and not require a custom constant?